services/shhext/chat/encryption_multi_device_test.go

```diff
@@ -1,121 +1,111 @@
 package chat
 
 import (
+	"crypto/ecdsa"
+	"fmt"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/stretchr/testify/suite"
 	"os"
 	"sort"
 	"testing"
 )
 
+const (
+	aliceUser = "alice"
+	bobUser   = "bob"
+)
+
 func TestEncryptionServiceMultiDeviceSuite(t *testing.T) {
 	suite.Run(t, new(EncryptionServiceMultiDeviceSuite))
 }
 
+type serviceAndKey struct {
+	encryptionServices []*EncryptionService
+	key                *ecdsa.PrivateKey
+}
+
 type EncryptionServiceMultiDeviceSuite struct {
 	suite.Suite
-	alice1 *EncryptionService
-	bob1   *EncryptionService
-	alice2 *EncryptionService
-	bob2   *EncryptionService
-	alice3 *EncryptionService
+	services map[string]*serviceAndKey
 }
 
-func (s *EncryptionServiceMultiDeviceSuite) SetupTest() {
-	const (
-		aliceDBPath1 = "/tmp/alice1.db"
-		aliceDBKey1  = "alice1"
-		aliceDBPath2 = "/tmp/alice2.db"
-		aliceDBKey2  = "alice2"
-		aliceDBPath3 = "/tmp/alice3.db"
-		aliceDBKey3  = "alice3"
-		bobDBPath1   = "/tmp/bob1.db"
-		bobDBKey1    = "bob1"
-		bobDBPath2   = "/tmp/bob2.db"
-		bobDBKey2    = "bob2"
-	)
-
-	os.Remove(aliceDBPath1)
-	os.Remove(bobDBPath1)
-	os.Remove(aliceDBPath2)
-	os.Remove(bobDBPath2)
-	os.Remove(aliceDBPath3)
-
-	alicePersistence1, err := NewSQLLitePersistence(aliceDBPath1, aliceDBKey1)
+func setupUser(user string, s *EncryptionServiceMultiDeviceSuite, n int) {
+	key, err := crypto.GenerateKey()
 	if err != nil {
 		panic(err)
 	}
 
-	alicePersistence2, err := NewSQLLitePersistence(aliceDBPath2, aliceDBKey2)
-	if err != nil {
-		panic(err)
+	s.services[user] = &serviceAndKey{
+		key:                key,
+		encryptionServices: make([]*EncryptionService, n),
 	}
 
-	alicePersistence3, err := NewSQLLitePersistence(aliceDBPath3, aliceDBKey3)
-	if err != nil {
-		panic(err)
-	}
+	for i := 0; i < n; i++ {
+		installationID := fmt.Sprintf("%s%d", user, i+1)
+		dbPath := fmt.Sprintf("/tmp/%s.db", installationID)
 
-	bobPersistence1, err := NewSQLLitePersistence(bobDBPath1, bobDBKey1)
-	if err != nil {
-		panic(err)
-	}
+		os.Remove(dbPath)
 
-	bobPersistence2, err := NewSQLLitePersistence(bobDBPath2, bobDBKey2)
-	if err != nil {
-		panic(err)
-	}
+		persistence, err := NewSQLLitePersistence(dbPath, "key")
+		if err != nil {
+			panic(err)
```

Comment:
I would avoid `panic` in tests. You can just return `error` and assert it in a setup method.